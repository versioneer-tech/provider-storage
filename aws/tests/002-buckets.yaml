---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  labels:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
  name: alice-002
  namespace: storage
spec:
  buckets:
  - bucketName: alice-002
    discoverable: false
  - bucketName: alice-shared-002
    discoverable: true
  owner: alice-002
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-alice-002, bucket-alice-002, bucket-alice-shared-002,
      and 4 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-alice-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    userRef:
      name: alice-002
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: alice-002
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-alice-shared-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-shared-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002.owner.alice-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::alice-002",
              "arn:aws:s3:::alice-002/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-owner-alice-shared-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002.owner.alice-shared-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::alice-shared-002",
              "arn:aws:s3:::alice-shared-002/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-owner-alice-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002.owner.alice-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-002.owner.alice-002
    userRef:
      name: alice-002
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-owner-alice-shared-002
  labels:
    crossplane.io/composite: alice-002
  name: alice-002.owner.alice-shared-002
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: alice-002.owner.alice-shared-002
    userRef:
      name: alice-002
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-alice-002
  generateName: alice-002-
  labels:
    crossplane.io/composite: alice-002
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: alice-002
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
