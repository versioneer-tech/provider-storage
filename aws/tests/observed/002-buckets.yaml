---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-002-alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: s3.aws.upbound.io/v1beta2
kind: Bucket
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: bucket-alice-002-alice-shared-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: policy-alice-002.owner.alice-shared-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-002.owner.alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: user-policy-attachement-alice-002.owner.alice-shared-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/claim-name: alice-002
    crossplane.io/claim-namespace: default
    crossplane.io/composition-resource-name: access-key-alice-002
spec: {}
status:
  conditions:
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: Available
    status: "True"
    type: Ready
  - lastTransitionTime: "1970-01-01T00:00:00Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
