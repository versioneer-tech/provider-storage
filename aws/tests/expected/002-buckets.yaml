# Copyright 2025, EOX (https://eox.at) and Versioneer (https://versioneer.at)
# SPDX-License-Identifier: Apache-2.0

---
apiVersion: pkg.internal/v1beta1
kind: Storage
metadata:
  name: s-jeff
spec:
  bucketAccessGrants:
  - bucketName: s-jeff-shared
    grantedAt: "2025-09-29T10:15:00Z"
    grantee: joe
    permission: ReadOnly
  bucketAccessRequests:
  - bucketName: s-joe
    reason: Need access
    requestedAt: "2025-09-29T10:10:00Z"
  buckets:
  - bucketName: s-jeff
  - bucketName: s-jeff-shared
    discoverable: true
  principal: jeff
status:
  conditions:
  - lastTransitionTime: "2024-01-01T00:00:00Z"
    message: 'Unready resources: access-key-jeff, bucket-s-jeff, bucket-s-jeff-shared,
      and 7 more'
    reason: Creating
    status: "False"
    type: Ready
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: access-key-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    userRef:
      name: jeff
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
  writeConnectionSecretToRef:
    name: jeff
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: s-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: s3.aws.m.upbound.io/v1beta1
kind: Bucket
metadata:
  annotations:
    crossplane.io/composition-resource-name: bucket-s-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    region: eu-central-1
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: kubernetes.m.crossplane.io/v1alpha1
kind: Object
metadata:
  annotations:
    crossplane.io/composition-resource-name: observe-jeff-s-joe
  labels:
    crossplane.io/composite: s-jeff
  name: observe-jeff-s-joe
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    manifest:
      apiVersion: iam.aws.m.upbound.io/v1beta1
      kind: Policy
      metadata:
        name: jeff.s-joe
        namespace: default
  managementPolicies:
  - Observe
  providerConfigRef:
    kind: ProviderConfig
    name: provider-kubernetes
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-joe-s-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: joe.s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:ListBucket"
            ],
            "Resource": [
              "arn:aws:s3:::s-jeff-shared"
            ]
          },
          {
            "Effect": "Allow",
            "Action": [
              "s3:GetObject"
            ],
            "Resource": [
              "arn:aws:s3:::s-jeff-shared/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-s-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: jeff.s-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::s-jeff",
              "arn:aws:s3:::s-jeff/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: Policy
metadata:
  annotations:
    crossplane.io/composition-resource-name: policy-s-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: jeff.s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Effect": "Allow",
            "Action": [
              "s3:*"
            ],
            "Resource": [
              "arn:aws:s3:::s-jeff-shared",
              "arn:aws:s3:::s-jeff-shared/*"
            ]
          }
        ]
      }
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-s-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: jeff.s-jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: jeff.s-jeff
    userRef:
      name: jeff
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: UserPolicyAttachment
metadata:
  annotations:
    crossplane.io/composition-resource-name: upa-s-jeff-shared
  labels:
    crossplane.io/composite: s-jeff
  name: jeff.s-jeff-shared
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider:
    policyArnRef:
      name: jeff.s-jeff-shared
    userRef:
      name: jeff
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
---
apiVersion: iam.aws.m.upbound.io/v1beta1
kind: User
metadata:
  annotations:
    crossplane.io/composition-resource-name: user-jeff
  labels:
    crossplane.io/composite: s-jeff
  name: jeff
  namespace: default
  ownerReferences:
  - apiVersion: pkg.internal/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Storage
    name: s-jeff
    uid: ""
spec:
  forProvider: {}
  providerConfigRef:
    kind: ProviderConfig
    name: provider-aws
